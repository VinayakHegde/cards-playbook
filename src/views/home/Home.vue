<script>
import Form from "@/components/form/Form.vue";
export default {
  components: {
    Form
  }
};
</script>
<template>
  <main>
    <Form />
  </main>
</template>
