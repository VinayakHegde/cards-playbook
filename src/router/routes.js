export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
