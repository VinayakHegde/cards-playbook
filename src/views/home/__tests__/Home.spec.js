import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import Home from "../Home.vue";
import routes from "@/router/routes";

const localVue = createLocalVue();
localVue.use(VueRouter);
describe("Home.vue", () => {
  it("match the snapshot", () => {
    const router = new VueRouter({ routes });
    const wrapper = mount(Home, {
      localVue,
      router
    });
    expect(wrapper).toMatchSnapshot();
  });
});
