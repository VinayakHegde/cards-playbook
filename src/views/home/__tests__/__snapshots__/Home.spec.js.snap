// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue match the snapshot 1`] = `
<main>
  <form autocomplete="off" class="form">
    <div class="fields">
      <fieldset class="field">
        <legend>Card 1</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 2</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 3</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 4</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 5</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 6</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 7</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 8</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 9</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
      <fieldset class="field">
        <legend>Card 10</legend> <input placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off">
      </fieldset>
    </div>
    <div class="single-field"><label for="field-rotation">Rotation Card</label> <input id="field-rotation" placeholder="Enter card" minlength="2" maxlength="3" autocomplete="off"></div> <button type="submit">Submit</button>
  </form>
</main>
`;
